src/main/java/com/example/demo/services/UsersService.java

```diff
@@ -125,12 +141,382 @@ public Integer extractData(ResultSet rs) throws SQLException, DataAccessExceptio
 					userBasicDetailsBO.setReligion(rs.getString("RELIGION"));
 					userBasicDetailsBO.setCaste(rs.getString("CASTE"));
 					userBasicDetailsBO.setNationality(rs.getString("NATIONALITY"));
+					
+					//commenting this out as there is no column as GENDER in user_basic_details
 					userBasicDetailsBO.setGender(rs.getString("GENDER"));
+					
+					//commenting this out as there is no column as ALTERNATE_MOBILE in user_basic_details
 					userBasicDetailsBO.setAlternateMobile(rs.getString("ALTERNATE_MOBILE"));
 					userBasicDetailsBOs.add(userBasicDetailsBO);
 				}
 				return userBasicDetailsBOs;
 			}
 		});
 	}
+	
+	public UserBasicDetailsBO getBasicDetailsOfOneUser(String userId)
+	{
+		String query = "SELECT * FROM USER_BASIC_DETAILS where user_id=?";
+		
+		return jdbcTemplate.query(query, new PreparedStatementSetter() {
+			
+			@Override
+			public void setValues(PreparedStatement ps) throws SQLException {
+				ps.setString(1, userId);
+			}
+			
+		}, new ResultSetExtractor<UserBasicDetailsBO>() {
+			
+			@Override
+			public UserBasicDetailsBO extractData(ResultSet rs) throws SQLException, DataAccessException {
+				
+				UserBasicDetailsBO userBasicDetailsBO = new UserBasicDetailsBO(); 
```

Comment:
Instead of populating userBasicDetailsBO you can directly put all this data in AdvanceDetalsBO as it extends userBasicDetailsBO